EE.C

```diff
@@ -0,0 +1,478 @@
+			      }
+			      else Nnon_bjet ++;
+			  }
+		      }
+
+		  }
+	     // }
+
+	      //if(Njet == 0){cout<<"entry : "<<jentry << endl;}
+	     // if(Lep_pt[0] == -999) continue;
+
+
+	      if(Njet < 2) continue;//Number of jet >=2
+
+		 //cout<<"NJet: "<<Njet<<endl; 
+	      njet_DY->Fill(Njet);
+	      nbjet_DY->Fill(Nbjet);
+
+	      njvsnbj_DY->Fill(Nnon_bjet,Nbjet);
+
+	      //if(Nbjet < 1) continue;// >= 1 b-jet
+
+	      for(int i=0; i<nJets; i++){
+		  if(jet_eta[i] == -999.) continue;
+		  if(jet_phi[i] == -999.) continue;
+		  if(jet_pt[i] <= 30) continue; //jet pT > 30GeV
+		  if(jet_eta[i] >= 5.0) continue; // jet |eta| < 5.0
+		  
+		  
+		  //lepton
+		  Lep_pt[0] = lep_pt[0];
+		  Lep_pt[1] = lep_pt[1];
+		  
+		  Lep_eta[0] = lep_eta[0];
+		  Lep_eta[1] = lep_eta[1];
+		  
+		  Lep_phi[0] = lep_phi[0];
+		  Lep_phi[1] = lep_phi[1];
+		  
+		  Lep_Iso[1] = lep_pfIso[0];
+		  Lep_Iso[1] = lep_pfIso[1];
+
+		  Lep_E[0] = lep_E[0];
+		  Lep_E[1] = lep_E[1];
+
+		  //di-lepton(e,mu)
+		  Dil_dPhi[0] = dil_dPhi;
+		  Dil_dR[0] = dil_dR;
+		  Dil_mass[0] = dil_mass;
+
+		  //Missing ET
+		  MET[0] = met_pt;
+		  
+		  //jet
+		  Jet_pt.push_back(jet_pt[i]);
+		  Jet_eta.push_back(jet_eta[i]);
+		  Jet_phi.push_back(jet_phi[i]);
+          Jet_E.push_back(jet_E[i]);
+		  }
+	      
+	      
+	      if(Lep_pt[0] == -999) continue;
+	      
+	  
+	  //pT
+	  float pt1 = Lep_pt[0];
+	  float pt2 = Lep_pt[1];
+
+
+	  leppT_DY->Fill(pt1);
+	  leppT_DY->Fill(pt2);
+
+	  //Iso
+		  eleIso_DY->Fill(Lep_Iso[1]);
+		  eleIso_DY->Fill(Lep_Iso[0]);
+	 
+
+	  //e-mu dPhi and dR, e-mu mass
+	  dPhi_DY->Fill(Dil_dPhi[0]);
+	  dR_DY->Fill(Dil_dR[0]);
+	  ee_mass_DY->Fill(Dil_mass[0]);
+
+	  //eta
+	  float eta1 = Lep_eta[0];
+	  float eta2 = Lep_eta[1];
+
+
+	  lepeta_DY->Fill(eta1);
+	  lepeta_DY->Fill(eta2);
+
+	  //phi
+	  float phi1 = Lep_phi[0];
+	  float phi2 = Lep_phi[1];
+
+
+	  lepphi_DY->Fill(phi1);
+	  lepphi_DY->Fill(phi2);
+
+	 //leading and 2nd leading declaration 
+	  float leadjet_pT = -999;
+	  float secjet_pT = -999;
+
+	  //dR(electron,b-jet)..what is closest electron?
+	  //float EleBjet_dR;
+
+//		    btagEvent_DY->Fill(lep_pt[0]);
+//		    btagEvent_DY->Fill(lep_pt[1]);
+
+		    //jet pT and jet eta(if Number of b-jet = 1, it is possible to use)
+		    for(int k=0; k<Jet_pt.size(); k++){
+			
+		    jetpT_DY->Fill(Jet_pt[k]);
+		    jeteta_DY->Fill(Jet_eta[k]);
+		    }
+
+
+		    //all jet pT sum^2
+		    float sum_jetpT = 0.;
+
+		    //leading jet
+		    for(int j=0; j<bJet_pt.size(); j++){
+		        bjetpT_DY->Fill(bJet_pt[j]); // all b jet pT
+		        bjeteta_DY->Fill(bJet_eta[j]); // all b jet pT
+
+			//all jet pT sum
+			sum_jetpT = sum_jetpT + sqrt(Jet_pt[j]*jet_pt[j]);
+
+
+			//b-jet pT vs dR
+			float dphi = Lep_phi[1] - bJet_phi[j];
+			float deta = Lep_eta[1] - bJet_eta[j];
+			if(dphi < -TMath::Pi()) dphi = dphi + 2.*TMath::Pi();
+			if(dphi >= TMath::Pi()) dphi = dphi - 2.*TMath::Pi();
```

Comment:
Using TLorentzVectors is your friend there, as it contains https://root.cern.ch/doc/master/classTLorentzVector.html#ad40f81b4495bd10496ebcb78b88f4082 and also DeltaR capacities that automatically do this for you.